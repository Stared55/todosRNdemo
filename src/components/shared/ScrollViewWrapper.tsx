import * as React from 'react';
import { StyleSheet } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { Colors } from 'src/utils/styles/colors';
import { yScale } from 'src/utils/styles/scale';
import { ViewStyles } from 'src/utils/types/styleTypes';

export interface ScrollViewWrapperProps {
  children: React.ReactNode;
  scrollViewWrapper?: ViewStyles;
  contentContainer?: ViewStyles;
}

const ScrollViewWrapper: React.SFC<ScrollViewWrapperProps> = ({
  children,
  scrollViewWrapper,
  contentContainer,
}) => {
  return (
    <ScrollView
      style={{ ...styles.scrollViewWrapper, ...scrollViewWrapper }}
      contentContainerStyle={{
        ...styles.contentContainer,
        ...contentContainer,
      }}>
      {children}
    </ScrollView>
  );
};

export default ScrollViewWrapper;

const styles = StyleSheet.create({
  contentContainer: {
    alignItems: 'center',
    paddingTop: yScale(12),
    justifyContent: 'space-between',
  },
  scrollViewWrapper: {
    flex: 1,
    backgroundColor: Colors.white,
  },
});
