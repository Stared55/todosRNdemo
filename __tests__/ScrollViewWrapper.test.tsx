import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ScrollViewWrapper from 'src/components/shared/ScrollViewWrapper';
import { Text } from 'react-native';

describe('ScrollViewWrapper', () => {
  let component: ShallowWrapper;
  let props: any;
  let children = <Text>test test</Text>;
  const scrollViewWrapper = { backgroundColor: 'green' };
  const contentContainer = { backgroundColor: 'yellow' };

  beforeEach(() => {
    component = shallow(
      <ScrollViewWrapper
        scrollViewWrapper={scrollViewWrapper}
        contentContainer={contentContainer}>
        {children}
      </ScrollViewWrapper>,
    );
    props = component.props();
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correct children', () => {
    const c = component.find({ children: children });
    expect(c).toHaveLength(1);
  });

  it('should render correct container styles', () => {
    expect(props.style.backgroundColor).toEqual('green');
  });

  it('should render correct label styles', () => {
    expect(props.contentContainerStyle.backgroundColor).toEqual('yellow');
  });
});
