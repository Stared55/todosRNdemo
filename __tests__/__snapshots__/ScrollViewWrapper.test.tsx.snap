// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollViewWrapper should match to snapshot 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "yellow",
      "justifyContent": "space-between",
      "paddingTop": 23.541176470588233,
    }
  }
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
    }
  }
>
  <Text>
    test test
  </Text>
</ScrollView>
`;
